plansys2_core/include/plansys2_core/Types.hpp

```diff
@@ -39,39 +39,58 @@ typedef std::vector<PlanItem> Plan;
 class Instance : public plansys2_msgs::msg::Param
 {
 public:
+  Instance()
+  : plansys2_msgs::msg::Param() {}
   explicit Instance(const std::string & name, const std::string & type = {})
   : plansys2_msgs::msg::Param(parser::pddl::fromStringParam(name, type)) {}
-  explicit Instance(const plansys2_msgs::msg::Param & instance)
+  Instance(const plansys2_msgs::msg::Param & instance)
   : plansys2_msgs::msg::Param(instance) {}
 };
 
 class Predicate : public plansys2_msgs::msg::Node
 {
 public:
+  Predicate()
+  : plansys2_msgs::msg::Node() {}
   explicit Predicate(const std::string & pred)
   : plansys2_msgs::msg::Node(parser::pddl::fromStringPredicate(pred)) {}
-  explicit Predicate(const plansys2_msgs::msg::Node & pred)
+  Predicate(const plansys2_msgs::msg::Node & pred)
   : plansys2_msgs::msg::Node(pred) {}
 };
 
 class Function : public plansys2_msgs::msg::Node
 {
 public:
+  Function()
+  : plansys2_msgs::msg::Node() {}
   explicit Function(const std::string & func)
   : plansys2_msgs::msg::Node(parser::pddl::fromStringFunction(func)) {}
-  explicit Function(const plansys2_msgs::msg::Node & func)
+  Function(const plansys2_msgs::msg::Node & func)
   : plansys2_msgs::msg::Node(func) {}
 };
 
 class Goal : public plansys2_msgs::msg::Tree
 {
 public:
+  Goal()
+  : plansys2_msgs::msg::Tree() {}
   explicit Goal(const std::string & goal)
   : plansys2_msgs::msg::Tree(parser::pddl::fromString(goal)) {}
-  explicit Goal(const plansys2_msgs::msg::Tree & goal)
+  Goal(const plansys2_msgs::msg::Tree & goal)
```

Comment:
```suggestion
  explicit Goal(const plansys2_msgs::msg::Tree & goal)
```
All of the additions of `explicit` to this file in the original PR were inspired by running cpplint on this package via ament_lint_auto:
```
- cpplint
  <<< failure message
    -- run_test.py: invoking following command in '/home/axydes/src/upstream-plansys2-test/src/third_party/plansys2/ros2_planning_system/plansys2_core':
     - /opt/ros/foxy/bin/ament_cpplint --xunit-file /home/axydes/src/upstream-plansys2-test/build/plansys2_core/test_results/plansys2_core/cpplint.xunit.xml
    /home/axydes/src/upstream-plansys2-test/src/third_party/plansys2/ros2_planning_system/plansys2_core/include/plansys2_core/Types.hpp:46:  Single-parameter constructors should be marked explicit.  [runtime/explicit] [5]
    /home/axydes/src/upstream-plansys2-test/src/third_party/plansys2/ros2_planning_system/plansys2_core/include/plansys2_core/Types.hpp:57:  Single-parameter constructors should be marked explicit.  [runtime/explicit] [5]
    /home/axydes/src/upstream-plansys2-test/src/third_party/plansys2/ros2_planning_system/plansys2_core/include/plansys2_core/Types.hpp:68:  Single-parameter constructors should be marked explicit.  [runtime/explicit] [5]
    /home/axydes/src/upstream-plansys2-test/src/third_party/plansys2/ros2_planning_system/plansys2_core/include/plansys2_core/Types.hpp:79:  Single-parameter constructors should be marked explicit.  [runtime/explicit] [5]
    Category 'runtime/explicit' errors found: 4
    Total errors found: 4
```

Reply:
See commit 25dc11d.